.github/workflows/.windows.yml

```diff
@@ -152,7 +155,10 @@ jobs:
       -
         name: Build base image
         run: |
-          docker pull ${{ env.WINDOWS_BASE_IMAGE }}:${{ env.WINDOWS_BASE_IMAGE_TAG }}
+          $null = docker images -q "${{ env.WINDOWS_BASE_IMAGE }}:${{ env.WINDOWS_BASE_IMAGE_TAG }}" 2>&1
+          if ($LASTEXITCODE -ne 0) {
+            docker pull ${{ env.WINDOWS_BASE_IMAGE }}:${{ env.WINDOWS_BASE_IMAGE_TAG }}
+          }
           docker tag ${{ env.WINDOWS_BASE_IMAGE }}:${{ env.WINDOWS_BASE_IMAGE_TAG }} microsoft/windowsservercore
           docker build --build-arg GO_VERSION -t ${{ env.TEST_IMAGE_NAME }} -f Dockerfile.windows .
```

Comment:
Wondering; I think we're now always using the same base image for all windows flavours, correct? (i.e., `env.WINDOWS_BASE_IMAGE` is always the same?)

Should we go back to updating the Dockerfile to not use `microsoft/windowsservercore` (which requires manually tagging that name before building), and just use `mcr.microsoft.com/windows/servercore` and a default tag (`ltsc2022` ?) in the Dockerfile?

That way `docker build` would default to use the locally available image if it's present.

I _think_ backwards compatibility also improved, so `mcr.microsoft.com/windows/servercore:ltsc2022` may even work on 2019?